app/controllers/posts_controller.rb

```diff
@@ -2,8 +2,10 @@ class PostsController < ApplicationController
   def index
     if params[:user_id].present?
       # List posts of a specific user
+
       @user = User.find(params[:user_id])
-      @posts = @user.posts
+      # @posts = @user.posts
+      @posts = Post.includes(:author).where(author: params[:user_id])
```

Comment:
- [ ] To solve the N+1 problem you're supposed to get the comments when fetching the posts, not the author.
you can archive this by writing this line like: `@posts = @user.posts.includes(:comments)`

Reply:
thanks @mikethreels for the help